src/pyclaw/sharpclaw/solver.py

```diff
@@ -652,10 +655,12 @@ def get_dt_new(self):
                 # Step-size update of starting methods
                 sspcoeff_ratio = self.sspcoeff0/self.sspcoeff
                 self.dt = sspcoeff_ratio * self.dt * self.cfl_desired / cfl
-                if self.time_integrator == 'SSPLMMk3' and self.check_lmm_cond:
+                if self.time_integrator == 'SSPLMMk3' and self.check_lmm_cond and not self.lmm_cond:
```

Comment:
What is `lmm_cond`?  How does it differ from `check_lmm_cond`?  Can these things be given clearer names?